### PR TITLE
add dice loss and bce loss

### PR DESCRIPTION
**Pull request by wuyefeilin**

### CODE REVIEW

File: pdseg/models/model_builder.py [hunk 1]
```diff
@@ -144,11 +157,35 @@ def build_model(main_prog, start_prog, phase=ModelPhase.TRAIN):
...
+                if "dice" in loss_type:
+                    avg_loss_list.append(multi_dice_loss(logits, label, mask))
+                    loss_valid = True
+                if "bce" in loss_type:
+                    avg_loss_list.append(multi_bce_loss(logits, label, mask))
+                    loss_valid = True
+                if not loss_valid:
+                    raise Exception("SOLVER.LOSS: {} is set wrong. it should "
+                            "include one of (softmax_loss, bce_loss, dice_loss) at least"
+                            " example: 'softmax_loss', 'dice_loss', bce_loss,dice_loss'")
```
[the remark this answers is not in the document]
Comment (wuyefeilin) [reply pdseg/models/model_builder.py]: 已更正

File: pdseg/loss.py [hunk 2]
```diff
@@ -48,6 +48,42 @@ def softmax_with_loss(logit, label, ignore_mask=None, num_classes=2):
...
+        logit, dim=reduce_dim) + fluid.layers.reduce_sum(
+        label, dim=reduce_dim)
+    dice_score = 1 - inse * 2 / (dice_denominator + epsilon)
+    label.stop_gradient = True
+    ignore_mask.stop_gradient = True
+    return fluid.layers.reduce_mean(dice_score)
+
+def bce_loss(logit, label, ignore_mask=None):
+    if logit.shape[1] != 1 or label.shape[1] != 1 or ignore_mask.shape[1] != 1:
+        raise Exception("dice loss is only applicable to binary classfication")
```
[the remark this answers is not in the document]
Comment (wuyefeilin) [reply pdseg/loss.py]: 已更正：

File: docs/configs/solver_group.md [hunk 3]
```diff
@@ -121,4 +121,18 @@ L2正则化系数
...
-<br/>
\ No newline at end of file
+<br/>
+
+## `loss`
+
+训练时选择的损失函数， 支持`softmax_loss`, `dice_loss`, `bce_loss`三种损失函数。其中`dice_loss`和`bce_loss`仅在两类分割问题中适用，`softmax_loss`不能与`dice_loss`
+或`bce_loss`组合，`dice_loss`可以和`bce_loss`组合使用。使用示例如下：
+
+`"softmax_loss"`或`"dice_loss,bce_loss"`
```
[the remark this answers is not in the document]
Comment (wuyefeilin) [reply docs/configs/solver_group.md]: 这边考虑了用户拼写错误的问题，使用列表容易对不上。通过字符串，利用in判断，感觉鲁棒性会好一些
